cpp/lib/src/MARC.cc

```diff
@@ -1592,6 +1592,31 @@ bool Record::addSubfield(const Tag &field_tag, const char subfield_code, const s
     return true;
 }
 
+bool Record::addSubfieldCreateFieldUnique(const Tag &field_tag, const char subfield_code, const std::string &subfield_value) {
+    const auto &field(findTag(field_tag));
+    bool success = true;
+    if (field == fields_.end()) {
+        success = insertField(field_tag, std::string(1, subfield_code) + subfield_value);
```

Comment:
please avoid passing a string directly (e.g. due to missing indicators)

`insertField(field_tag, {{subfield_code, subfield_value}})`